### ISSUE TITLE
Argon2 support

### ISSUE THREAD

**Issue by chittti** [weidai11/cryptopp#399]
Thanks for your hard work on this library!

Argon2 winning the [PHC](https://password-hashing.net/) made it a great choice for a memory hard KDF. BLAKE2 is already implemented here so I think it would be a bit less work than a fresh implementation. 

I've found some useful writeups [here](https://wiki.mozilla.org/NSS/ARGON2) although there doesn't seem to be any work in the past couple months.

**Comment by noloader** [comment]
Yeah, the memory hard KDFs are on our [roadmap](https://www.cryptopp.com/wiki/Roadmap).

**Comment by Bweyer6** [comment]
I see that Scrypt has since been added, but are there any future plans for implementing Argon2?

**Comment by noloader** [comment]
@Bweyer6,

I looked into Argon2 a while ago. The problem was/is, the code is very complex and I have not mustered the energy to untangle it.
